membership-common/src/main/scala/com/gu/memsub/subsv2/reads/CatJsonReads.scala

```diff
@@ -1,91 +1,64 @@
-      (__ \ "specificBillingPeriod").readNullable[Int] and
-      (__ \ "model").read[String] and
-      (__ \ "name").read[String] and
-      (__ \ "type").read[String] and
-      (__ \ "endDateCondition").read[EndDateCondition] and
-      (__ \ "upToPeriods").readNullable[Int] and
-      (__ \ "upToPeriodsType").readNullable[UpToPeriodsType]
-  )(ZuoraCharge.apply(_, _, _, _, _, _, _, _, _, _))
-
-  implicit val ProductReads = new Reads[Benefit] {
-    override def reads(json: JsValue): JsResult[Benefit] = json match {
-      case JsString(id) => Benefit.fromId(id).fold[JsResult[Benefit]](JsError(s"Bad product $id"))(e => JsSuccess(e))
+  private implicit val productRatePlanChargeProductTypeReads: Reads[ProductRatePlanChargeProductType] = new Reads[ProductRatePlanChargeProductType] {
+    override def reads(json: JsValue): JsResult[ProductRatePlanChargeProductType] = json match {
+      case JsString(id) =>
+        ProductRatePlanChargeProductType.fromId(id).fold[JsResult[ProductRatePlanChargeProductType]](JsError(s"Bad product $id"))(e => JsSuccess(e))
       case a => JsError(s"Malformed product JSON, needed a string but got $a")
     }
   }
 
-  implicit val catalogZuoraPlanBenefitReads: Reads[(ProductRatePlanChargeId, Benefit)] = (
+  private implicit val productRatePlanChargeReads: Reads[(ProductRatePlanChargeId, ProductRatePlanChargeProductType)] = (
     (__ \ "id").read[String].map(ProductRatePlanChargeId) and
-      (__ \ "ProductType__c").read[Benefit]
+      (__ \ "ProductType__c").read[ProductRatePlanChargeProductType]
   )(_ -> _)
 
-  implicit val listOfProductsReads = new Reads[Map[ProductRatePlanChargeId, Benefit]] {
-    override def reads(json: JsValue): JsResult[Map[ProductRatePlanChargeId, Benefit]] = json match {
+  private val productRatePlanChargesReads = new Reads[Map[ProductRatePlanChargeId, ProductRatePlanChargeProductType]] {
+    override def reads(json: JsValue): JsResult[Map[ProductRatePlanChargeId, ProductRatePlanChargeProductType]] = json match {
       case JsArray(vals) =>
         vals
-          .map(_.validate[(ProductRatePlanChargeId, Benefit)])
+          .map(_.validate[(ProductRatePlanChargeId, ProductRatePlanChargeProductType)])
           .filter(_.isSuccess)
           .toList // bad things are happening here, we're chucking away errors
-          .sequence[JsResult, (ProductRatePlanChargeId, Benefit)]
+          .sequence[JsResult, (ProductRatePlanChargeId, ProductRatePlanChargeProductType)]
           .map(_.toMap)
       case _ => JsError("No valid benefits found")
     }
   }
 
-  implicit val statusReads: Reads[Status] = new Reads[Status] {
-    override def reads(json: JsValue): JsResult[Status] = json match {
-      case JsString("Expired") => JsSuccess(Status.Legacy)
-      case JsString("Active") => JsSuccess(Status.Current)
-      case JsString("NotStarted") => JsSuccess(Status.Upcoming)
-      case a => JsError(s"Unknown status $a")
-    }
```

Comment:
we no longer do this checking - before, the implicit deserialisers would discard any plans that were not active in the catalog.  However since MDAPI is only working on existing subscriptions, we would only be looking things up rather than searching for one to create.